pkg/cgroup/resolve_container.go

```diff
@@ -40,30 +40,59 @@ type ContainerInfo struct {
 
 const (
 	unknownPath string = "unknown"
-
-	procPath   string = "/proc/%d/cgroup"
-	cgroupPath string = "/sys/fs/cgroup"
+	procPath    string = "/proc/%d/cgroup"
+	cgroupPath  string = "/sys/fs/cgroup"
 )
 
 var (
-	byteOrder binary.ByteOrder         = utils.DetermineHostByteOrder()
-	podLister kubelet.KubeletPodLister = kubelet.KubeletPodLister{}
+	instance *cache
+	once     sync.Once
+)
 
+type cache struct {
 	// map to cache data to speedup lookups
-	containerIDCache           = map[uint64]string{}
-	containerIDToContainerInfo = map[string]*ContainerInfo{}
-	cGroupIDToPath             = map[uint64]string{}
-
-	validContainerIDRegex         = regexp.MustCompile("^[a-zA-Z0-9]+$")
-	regexReplaceContainerIDPrefix = regexp.MustCompile(`.*//`)
-)
+	containerIDCache           map[uint64]string
+	containerIDToContainerInfo map[string]*ContainerInfo
+	cGroupIDToPath             map[uint64]string
+	mu                         sync.RWMutex
+}
 
 func Init() (*[]corev1.Pod, error) {
 	pods := []corev1.Pod{}
 	return &pods, nil
 }
 
+// GetConfig returns the singleton Config instance, creating it if necessary.
+func InitCache() {
+	once.Do(func() {
+		instance = newCache()
+	})
+}
```

Comment:
instead of `once.Do` how about we use `func init()` to initialise the cache? That will make `ensureCache...` unnecessary.

Reply:
I will have a look